### PR TITLE
Remove opentelemetry-sdk-extension-autoconfigure from samples 

### LINKED ISSUE #31886: [BUG] The unit tests of azure-core-metrics-opentelemetry failed when opentelemetry upgradding to 1.19.0
**Describe the bug**
Some unit tests of module `azure-core-metrics-opentelemetry` failed in the ci pipeline.

***Exception or Stack Trace***
The below log is from test class `com.azure.core.metrics.opentelemetry.MetricsTests`
```console
io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException: OTLP gRPC Metrics Exporter enabled but opentelemetry-exporter-otlp not found on classpath. Make sure to add it as a dependency to enable this feature.
```

**To Reproduce**
Upgrade to [opentelemetry](io.opentelemetry:opentelemetry-exporter-otlp) to 1.19.0, io.opentelemetry:opentelemetry-sdk-extension-autoconfigure to 1.19.0-alpha.
Or you can run the tests in branch [feature/spring-boot-3](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3)

***Code Snippet***
N/A

**Expected behavior**
Compatible with the opentelemetry to 1.19.0 and 1.19.0-alpha.

**Screenshots**
N/A

**Setup (please complete the following information):**
 - OS: [Windows 11]
 - IDE: [IntelliJ]
 - Library/Libraries: [com.azure:azure-core-amqp:current]
 - Java version: [17] 

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #31886 and removes confusing error messages from pipeline logs

### CODE REVIEW

File: sdk/core/azure-core-metrics-opentelemetry/pom.xml [hunk 1]
```diff
@@ -118,12 +118,6 @@
       <version>1.14.0</version>  <!-- {x-version-update;io.opentelemetry:opentelemetry-exporter-otlp;external_dependency} -->
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>io.opentelemetry</groupId>
-      <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
-      <version>1.14.0-alpha</version>  <!-- {x-version-update;io.opentelemetry:opentelemetry-sdk-extension-autoconfigure;external_dependency} -->
-      <scope>test</scope>
-    </dependency>
```
Comment (trask) [inline sdk/core/azure-core-metrics-opentelemetry/pom.xml:126]: just confirming that it's not possible to add a dependency only for `samples`?

**Review by trask — APPROVED** [review]

File: sdk/core/azure-core-metrics-opentelemetry/pom.xml (hunk 1 above)
Comment (lmolkova) [reply sdk/core/azure-core-metrics-opentelemetry/pom.xml:126]: @trask 
it should be possible to create samples-only project and then it should be fine. 
To be fair, I didn't push hard enough to keep it and solve global + autoconfigure problems in tests. If you believe we absolutely need to have it in samples, I can try harder :)

**Review by samvaity — APPROVED** [review]

File: sdk/core/azure-core-metrics-opentelemetry/pom.xml (hunk 1 above)
Comment (trask) [reply sdk/core/azure-core-metrics-opentelemetry/pom.xml:126]: no problem, I support this PR as-is, and thx for the tip about creating a samples-only project if we need to in the future

**Pull request merged**